### DISCUSSION TITLE
MongoDB with already existing data that shouldn't be duplicated

### DISCUSSION THREAD

**Discussion by ivan-ikl** [certtools/intelmq#2247] [Q&A] [answered]
Greetings,

I have a MongoDB collection which collects URLs, and has a lot of existing data that was not fetched using IntelMQ. Is it possible to configure the MongoDB output bot so that it checks whether a URL already exists in the collection, and if it does exist, avoids storing a new record?

Thank you in advance.

**Comment by sebix** [comment, marked as answer]
Would creating a [unique index](https://www.mongodb.com/docs/manual/core/index-unique/) in MongoDB work?

IntelMQ itself does (as of 2022-10-11) not have the ability to lookup values in MongoDB databases, but I agree that this would be a very good addition.
